dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/PLL/PLL.mo

```diff
@@ -0,0 +1,117 @@
+within Dynawo.Electrical.Controls.PLL;
+
+model PLL "Phase-Locked Loop"
+
+  import Modelica.Blocks;
+  import Modelica.ComplexBlocks;
+  import Modelica.Constants;
+    
+  import Dynawo.Connectors;
+  import Dynawo.Types;
+
+  parameter Types.PerUnit Kp "PLL voltage calculation proportional gain";
+  parameter Types.PerUnit Ki "PLL voltage calculation integrator gain";
+  parameter Types.PerUnit OmegaMinPu "Lower frequency limit (only positive values!)";
+  parameter Types.PerUnit OmegaMaxPu "Upper frequency limit";
+  parameter Types.PerUnit OmegaRefPu "Frequency refence value";
```

Comment:
Is this one supposed to be equals to 1 all the time?

Reply:
I looked it up again in the technical reference for the PLL from PowerFactory standard library. This value should be equal to the reference frequency from the reference machine. So it should be connected to the input omegaRefPu. 

Shall I implement in that way:
- delete: parameter OmegaRefPu "Frequency reference value";
- replace "OmegaRefPu" in: Blocks.Continuous.LimIntegrator I(outMax = OmegaMaxPu - OmegaRefPu, outMin = OmegaRefPu - OmegaMinPu) with "omegaRefPu.value"
- the input omegaRefPu is initialised with initial value Omega0Pu. This comes at the moment from the init model . Do you have another idea how to define it instead of giving it in the init model? (response to your previous comment)

Reply:
- delete parameter : ok

- replace OmegaRefPu with omegaRefPu.value : ok but you will probably need to build another block because the LimIntegrator expects its limits as parameters

- you can initialize omegaRefPu without init model because its start value is supposed to be 1 all the time. You can do the same as in the synchronous machine model:

`omegaRefPu(value(start = SystemBase.omegaRef0Pu))`

Is this fine for you?

Reply:
I deleted the parameter and initialised omegaRefPu as per your suggestion with the SystemBase value.

By now, regarding the limits of the integrator, I would like to replace the parameter OmegaRefPu by SystemBase.omegaRef0Pu, if it is fine for you:
- The limits for the integrator are practically not used since the limits are set very wide. Within these wide limits, the difference between frequency of the reference machine (or center of intertia) and the initial value (omegaRef0Pu) is negligiable. Therefore, the effect of the new block with dynamic limits will be negligiable as well.
- If the limits on LimIntegrator should be set more tight for some reason, an additional anti-wind-up might be necessary for proper angle and frequency detection of the PLL. We implemented the PLL according to PowerFactory standard library reference without anti-wind-up. 

With further development of the PLL, where tight limits of the frequency limits are needed, the adjustment could be done:
- replacement of LimIntegrator by a new block with inputs for limitation parameters (dynamic limits)
- consideration of omegaRefPu (input variable) for calculation of the limits for this new block

Reply:
ok to have SystemBase.omegaRef0Pu in the lim integrator!

Reply:
ok, fine!